pkg/analyzer/analyzer.go

```diff
@@ -169,6 +169,18 @@ func findNestedContext(pass *analysis.Pass, node ast.Node, stmts []ast.Stmt) *as
 			continue
 		}
 
+		// Ignore [context.Background] & [context.TODO].
+		if call, ok := assignStmt.Rhs[0].(*ast.CallExpr); ok {
+			if selector, ok := call.Fun.(*ast.SelectorExpr); ok {
+				ident, identOK := selector.X.(*ast.Ident)
+				if identOK &&
+					ident.Name == "context" &&
+					(selector.Sel.Name == "Background" || selector.Sel.Name == "TODO") {
+					continue
+				}
+			}
+		}
```

Comment:
@ldez would you mind giving this a look please? I'm unsure whether there's a better (simpler) way of ignoring `context.Background` and `context.TODO`.

Reply:
This is the right way.

I recommend creating a function to improve the maintenance and decrease the cyclomatic complexity:

```go

		// Ignore [context.Background] & [context.TODO].
		if isContextFunction(assignStmt.Rhs[0], "Background", "TODO") {
			continue
		}
```

```go
func isContextFunction(exp ast.Expr, fnName ...string) bool {
	call, ok := exp.(*ast.CallExpr)
	if !ok {
		return false
	}

	selector, ok := call.Fun.(*ast.SelectorExpr)
	if !ok {
		return false
	}

	ident, ok := selector.X.(*ast.Ident)
	if !ok {
		return false
	}

	return ident.Name == "context" && slices.Contains(fnName, ident.Name)
}
```

Reply:
Thanks!